backend/middleware/authentication.js

```diff
@@ -0,0 +1,63 @@
+var User = require("../model/user");
+var jwt = require("jsonwebtoken");
+var Device = require("../model/device");
+
+module.exports = {
+
+    checkForAlreadyUser: (req, res, next) => {
+        let body = req.body;
+        User.findOne({ email: body.email }, (err, obj) => {
+            if (err) {
+                res.status(500).json({ error: 1, message: "mongo db internel problem while saving the finding the user" });
+            } else {
+                if (obj == null) {
+                    next();
+                } else {
+                    res.status(200).json({ error: 1, message: "please provide unique email name " });
+                }
+            }
+        })
+    },
+
+    validateToken: (req, res, next) => {
+        const token = req.headers.token;
+        jwt.verify(token, "thanos", (err, decoded) => {
+            if (err) {
+                res.status(400).json({ error: 1, message: err.message });
+            } else {
+                req.id = decoded.id;
+                next();
+            }
+        })
+    },
+
+    checkForRepeatAction: (req, res, next) => {
```

Comment:
what is repeat? didnt get this

Reply:
sir , if user already inserted device and then he is again trying to insert then i will check here